src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1313,19 +1313,22 @@ def _match():
                 # TODO: can a DaemonDescription exist without a hostname?
                 raise err
 
-            if self.hostname == self.daemon_id:
-                # daemon_id == "hostname"
+            # use the bare hostname, not the FQDN.
```

Comment:
painful. I don't think that we'll get `DaemonDescription.service_id()` working reliably. 

Looks like we have to make this explicit, as we're doing the OSD - OSDSpec linkage.